src/mz390.java

```diff
@@ -1520,6 +1533,32 @@ private void compile_patterns(){
 		} catch (Exception e){
 			abort_error(4,"expression pattern error - " + e.toString());
 		}
+		/*
+		 * signed or unsigned decimal integer                                    // #509
+		 */
+		try {                                                                    // #509
+			signedDecimalPattern = Pattern.compile("([-+]?)([0-9]+)");           // #509			
+		} catch (PatternSyntaxException e) {                                     // #509
+			abort_error(5,"signed decimal pattern error - " + e.toString());     // #509
+		}                                                                        // #509
+		/*
+		 * binary, decimal and hexadecimal digits                                // #509
+		 */
+		try {                                                                    // #509			
+			binDigitsPattern = Pattern.compile("[01]+");                         // #509
+		} catch (PatternSyntaxException e) {                                     // #509
+			abort_error(6,"binary digits pattern error - " + e.toString());      // #509
+		}                                                                        // #509
+		try {                                                                    // #509			
+			decDigitsPattern = Pattern.compile("[0-9]+");                        // #509
+		} catch (PatternSyntaxException e) {                                     // #509
+			abort_error(7,"decimal digits pattern error - " + e.toString());     // #509
```

Comment:
Just a question: why are you using abort_error? Have you considered log_error?

Reply:
Because all the try-catch pattern compiles above the new ones used it.

Reply:
okay, fair. No problem.